File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] preArr = new int[nums.length];
        int[] postArr = new int[nums.length];
        int[] answer = new int[nums.length];

        //Array that is nums that mul before it
        for(int i = 0; i < nums.length; i++){
            if(i == 0){
                preArr[i] = 1;

            }
            else{
                preArr[i] = preArr[i-1] * nums[i-1];
            }
        }

        for(int i = nums.length - 1; i >= 0; i--){
            if(i == nums.length - 1){
                postArr[i] = 1;
            }
            else{
                postArr[i] = postArr[i + 1] * nums[i+1];
            }

        }

        for(int i = 0; i < nums.length; i++){
            answer[i] = preArr[i] * postArr[i];
        }
       
        return answer;
    }
}  

File: Data Structures & Algorithms/concatenation-of-array/submission-1.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        k = 0
        n = len(nums)
        ans = []
        for i in range(n * 2):
            if i < n:
                ans.append(nums[i])
            else:
                ans.append(nums[k])
                k +=1
        return ans
                

        